openssl/src/x509/store.rs

```diff
@@ -81,11 +88,53 @@ impl X509StoreBuilder {
 
 impl X509StoreBuilderRef {
     /// Adds a certificate to the certificate store.
-    // FIXME should take an &X509Ref
-    pub fn add_cert(&mut self, cert: X509) -> Result<(), ErrorStack> {
+    pub fn add_cert(&mut self, cert: &X509Ref) -> Result<(), ErrorStack> {
         unsafe { cvt(ffi::X509_STORE_add_cert(self.as_ptr(), cert.as_ptr())).map(|_| ()) }
     }
 
+    /// Sets the maximum verification depth, or the maximum number of intermediate CA certificates that can appear in a chain.
+    pub fn set_depth(&mut self, depth: c_int) -> Result<(), ErrorStack> {
+        unsafe { cvt(ffi::X509_STORE_set_depth(self.as_ptr(), depth)).map(|_| ()) }
+    }
+
+    cfg_if! {
+        if #[cfg(any(ossl102, libressl261))] {
+             /// Sets the verification flags used to verify a certificate against a chain.
+            pub fn set_flags(&mut self, flags: X509VerifyFlags) -> Result<(), ErrorStack> {
```

Comment:
Why would the argument type for this not be X509VerifyFlags on old versions?

Reply:
X509VerifyFlags is in the `verify` module which is only compiled in the top level `x509` module for `any(ossl102, libressl261)`. I could relax/remove the version condition to use X509VerifyFlags for older versions.

Reply:
If the type is used in APIs that exist in older versions, then it can be exposed on those older versions.

Reply:
Removed the top-level constraint on the `verify` module and added more fine-grained constraints on structs and bit flags in the `verify` module.